widgetssdk/src/main/java/com/glia/widgets/core/permissions/PermissionManager.kt

```diff
@@ -36,12 +37,17 @@ internal class PermissionManager(
     fun hasScreenSharingNotificationChannelEnabled(): Boolean =
         applicationContext.areNotificationsEnabledForChannel(NotificationFactory.NOTIFICATION_SCREEN_SHARING_CHANNEL_ID)
 
-    fun hasPermission(permission: String) = checkSelfPermission(applicationContext, permission) == PackageManager.PERMISSION_GRANTED
+    private fun hasPermission(permission: String) = checkSelfPermission(applicationContext, permission) == PackageManager.PERMISSION_GRANTED
```

Comment:
Eventually all permissions should be handled in the core, right?

Reply:
I don't think so.
The main reason is that it is not core responsibility, so in my opinion, the core must check permissions and not start media upgrades or screen sharing if necessary permission is not granted. Widgets is fully responsible for handling permissions. Also with the current implementation we have to override `public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults)` method in every activity to pass permissions results to core this is a headache in case, we use activity watcher.